src/components/MyAccount/Settings/AccountSettingsTab.tsx

```diff
@@ -34,56 +34,54 @@ const AccountSettingsTab = ({ settingsData }: { settingsData: Patron }) => {
 
   const submitAccountSettings = async (e) => {
     e.preventDefault()
+    setIsLoading(true)
     const payload = parsePayload(e.target, mostRecentPatronData)
-    try {
-      const response = await fetch(
-        `${BASE_URL}/api/account/settings/${mostRecentPatronData.id}`,
-        {
-          method: "PUT",
-          headers: {
-            "Content-Type": "application/json",
-          },
-          body: JSON.stringify(payload),
-        }
-      )
-      const responseData = await response.json()
-      if (response.ok) {
-        setMostRecentPatronData(updatePatronData(mostRecentPatronData, payload))
-        setCurrentlyEditing(false)
-        setModalProps(successModalProps)
-        openModal()
-      } else {
-        alert(`error: ${responseData}`)
-        setModalProps(failureModalProps)
-        openModal()
+    const response = await fetch(
+      `/research/research-catalog/api/account/settings/${mostRecentPatronData.id}`,
+      {
+        method: "PUT",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify(payload),
       }
-    } catch (error) {
-      alert("fetching error")
+    )
+    if (response.status === 200) {
+      setMostRecentPatronData((prevData) => updatePatronData(prevData, payload))
+      setCurrentlyEditing(false)
+      setModalProps(successModalProps)
+      openModal()
+    } else {
+      setModalProps(failureModalProps)
+      openModal()
     }
+    setIsLoading(false)
   }
 
-  return (
-    <Box className={styles.accountSettingsTab}>
+  return isLoading ? (
+    <SkeletonLoader showImage={false} />
```

Comment:
I was thinking of disabling the form and button but this is much nicer!

Reply:
Agreed this is really nice